draft-ietf-taps-transport-security.md

```diff
@@ -312,60 +263,6 @@ negotiate protocol options, such as cryptographic algorithms, and derive session
-to use an authenticated encryption with associated data (AEAD) algorithm, it also handles nonce
-generation and encoding for each record. The record protocol is hidden from the client behind a
-bytestream-oriented API.
-
-The handshake protocol serves several purposes, including: peer authentication, protocol option (key exchange
-algorithm and ciphersuite) negotiation, and key derivation. Peer authentication may be mutual; however, commonly,
-only the server is authenticated. X.509 certificates are commonly used in this authentication step, though
-other mechanisms, such as raw public keys {{?RFC7250}}, exist. The client is not authenticated unless explicitly
-requested by the server.
-
-The handshake protocol is also extensible. It allows for a variety of extensions to be included by either the client
-or server. These extensions are used to specify client preferences, e.g., the application-layer protocol to be driven
-with the TLS connection {{?RFC7301}}, or signals to the server to aid operation, e.g., Server Name Indication
-(SNI) {{?RFC6066}}. Various extensions also exist to tune the parameters of the record protocol, e.g., the
-maximum fragment length {{?RFC6066}} and record size limit {{?RFC8449}}.
-
-Alerts are used to convey errors and other atypical events to the endpoints. There are two classes of alerts: closure
-and error alerts. A closure alert is used to signal to the other peer that the sender wishes to terminate the connection.
-The sender typically follows a close alert with a TCP FIN segment to close the connection. Error alerts are used to
-indicate problems with the handshake or individual records. Most errors are fatal and are followed by connection
-termination. However, warning alerts may be handled at the discretion of the implementation.
-
-Once a session is disconnected all session keying material must be destroyed, with
-the exception of secrets previously established expressly for purposes of session resumption.
-TLS supports stateful and stateless resumption. (Here, "state" refers to bookkeeping on a per-session basis
-by the server. It is assumed that the client must always store some state information in order to resume a session.)
-
-### Security Features
-
-- Forward-secure session key establishment.
-- Cryptographic algorithm negotiation.
-- Stateful and stateless cross-connection session resumption.
-- Session caching and management.
-- Peer authentication (Certificate, raw public key, and pre-shared key).
-- Unilateral responder authentication.
-- Mutual authentication.
-- Application authentication delegation.
-- Record (channel) confidentiality and integrity.
-- Record replay prevention.
-- Application-layer feature negotiation.
-- Configuration extensions.
-- Early data support (starting with TLS 1.3).
-- Optional record-layer padding (starting with TLS 1.3).
-
-### Protocol Dependencies
-
-- In-order, reliable bytestream transport.
-- (Optionally) A PKI trust store for certificate validation.
-
```

Comment:
Agree to delete the detailed protocol descriptions - But did we really want to delete all the security features and protocol dependencies?
For the security features, probably yes, because we want people to read the RFCs/other sources to understand the protocols, and we don't want people to think that protocols with the same features are equivalent.
For the protocol dependencies, I'm less sure - we moved transport dependencies to a section below, but we lose some detail there, like "SRTP can run over TCP and UDP". And what about other dependencies, like Path MTU discovery or external key management? Could listing these dependencies still be useful for somebody implementing a Transport System?
Because I'm unsure, I'd consider keeping these protocol dependencies for now. But if you're sure, feel free to delete them.

Reply:
Yes, I think so. The dependencies are listed below, and the features aren't what we want to express to applications.